test/integration/spec/rtcpeerconnection.js

```diff
@@ -786,42 +735,133 @@ function makeStream(constraints) {
-    test.peerConnection.addStream(stream);
-    if (isFirefox) {
-      const expectedTracks = stream.getTracks();
-      const actualTracks = test.peerConnection.getLocalStreams()[0].getTracks();
-      assertMediaStreamTracksEqual(actualTracks, expectedTracks);
-      return;
-    }
-    assert.equal(test.peerConnection.getLocalStreams()[0], stream);
+  it('should add each of the MediaStreamTracks to the RTCPeerConnection', () => {
+    const senders = stream.getTracks().map(track => test.peerConnection.addTrack(track, stream));
+    const addedTracks = 'getSenders' in RTCPeerConnection.prototype
+      ? test.peerConnection.getSenders().map(sender => sender.track)
+      : flatMap(test.peerConnection.getLocalStreams(), stream => stream.getTracks());
+    assert.deepEqual(addedTracks, stream.getTracks());
+    assert.deepEqual(senders.map(sender => sender.track), stream.getTracks());
+  });
+}
+
+function testRemoveTrack() {
+  var test;
+  var senders;
+  var stream;
+
+  before(async () => {
+    stream = await makeStream();
   });
 
-  context('when adding a stream that is already added', () => {
-    it('should not throw an exception', () => {
-      test.peerConnection.addStream(stream);
-      assert.doesNotThrow(() => test.peerConnection.addStream(stream));
+  beforeEach(async () => {
+    test = await makeTest();
+    senders = stream.getTracks().map(track => test.peerConnection.addTrack(track, stream));
+  });
+
+  context('when the RTCPeerConnection is closed', () => {
+    var exception;
+
+    beforeEach(() => {
+      test.peerConnection.close();
+      try {
+        test.peerConnection.removeTrack(senders.find(sender => sender.track.kind === 'audio'));
+      } catch (e) {
+        exception = e;
+      }
     });
 
-    it('should not add the stream', () => {
-      test.peerConnection.addStream(stream);
-      test.peerConnection.addStream(stream);
-      assert.equal(test.peerConnection.getLocalStreams().length, 1);
+    it('should throw', () => {
+      assert(!!exception);
     });
+
+    it('should not remove the MediaStreamTrack', () => {
+      const presentTracks = new Set('getSenders' in RTCPeerConnection.prototype
+        ? test.peerConnection.getSenders().map(sender => sender.track)
+        : flatMap(test.peerConnection.getLocalStreams(), stream => stream.getTracks()));
+      assert(presentTracks.has(stream.getAudioTracks()[0]));
+    });
+  });
+
+  context('when the MediaStreamTrack is already removed from the RTCPeerConnection', () => {
+    var exception;
+
+    beforeEach(() => {
+      try {
+        test.peerConnection.removeTrack(senders.find(sender => sender.track.kind === 'audio'));
+      } catch (e) {
+        exception = e;
+      }
+    });
+
+    it('should not throw', () => {
+      assert(!exception);
+    });
+  });
+
+  it('should remove the MediaStreamTrack from the RTCPeerConnection', () => {
+    test.peerConnection.removeTrack(senders.find(sender => sender.track.kind === 'audio'));
```

Comment:
Can we either save a reference to the RTCRtpSender or do a `find` comparing on equality of `sender.track`? (E.g., saving a reference to the local audio MediaStreamTrack, if necessary.)